### PR TITLE
DOC-3423: Standardize "New in vXX.Y" version callouts

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes DOC-3423:

- All references to "New in vXX.Y" have been replaced with a Liquid tag
- Installed a new Vale guardrail to prevent people from using the wrong syntax

### CODE REVIEW

**Review by stbof — APPROVED** [review]

**Comment by stbof** [comment]
However, the style guide should be updated as well: 

https://github.com/cockroachdb/docs/blob/master/StyleGuide.md#version-tags

**Comment by stbof** [comment]
Also, you might want to add a snippet here: 

https://cockroachlabs.atlassian.net/wiki/spaces/ED/pages/2193490166/Code+editors+for+authoring+in+our+Markdown+dialect#Using-Atom-snippets

```
  'new':
    'prefix': 'new'
    'body': '{% include_cached new-in.html version="$1" %}'
```

File: StyleGuide.md [hunk 1]
```diff
@@ -757,21 +757,19 @@ Examples help show the feature in action. Examples follow a basic format:
...
 ~~~
-<span class="version-tag">New in vX.X:</span>
+{% include_cached new-in.html version="v22.1" %}
 ~~~
 
 Put version tags at the beginning of a paragraph, sentence, or description in a table cell.
 
-Note: Do not use version tags in bulleted lists items. To denote a new feature in a bulleted list, start the bulleted item with "**New in vX.X:**".
-
 If a feature is new in a GA release, use the major release number for the release version tag (e.g., `New in v22.1`).
```
Comment (stbof) [inline StyleGuide.md:768]: I think these examples in parens are confusing. Could you change them to use the Liquid version?

**Review by stbof — CHANGES_REQUESTED** [review]
See comment about examples in Style Guide.

File: StyleGuide.md (hunk 1 above)
Comment (nickvigilante) [reply StyleGuide.md]: Done.

**Review by stbof — APPROVED** [review]

**Pull request merged**